### ISSUE TITLE
[Feature Request] Command Prefix → Support for Repeated Letters

### ISSUE THREAD

**Issue by FelipeRearden** [tadashi-aikawa/obsidian-another-quick-switcher#180]
Hello @tadashi-aikawa !!!!

I am struggling to implement a workflow related to the feature `Command Prefix` that I would like to share with you. 

![78B00502-7B36-4459-8FF9-D0EEDDFA108E](https://github.com/tadashi-aikawa/obsidian-another-quick-switcher/assets/85180696/a42a45f4-faf3-416d-8338-5dfe5d9772c5)

I am gonna try to explain in my own words, please let me know if something is not clear 🙏

## MY WORKFLOW

I am gonna use a simple example to explain this... 

My worklfow is to **navigate** between some AQS modal using `Recent Search` as a starting point and then going to others AQS Modals using the feature `Command Prefix` as a hotkey.

### AQS Modals 

1️⃣ **Recent Search**
⦿  Hotkey <kbd>CMD+]</kbd>
⦿  Command Prefix `EMPTY`

2️⃣ **File name search**
⦿  Hotkey <kbd>EMPTY</kbd>
⦿  Command Prefix `]`

3️⃣ **Landmark search**
⦿  Hotkey <kbd>EMPTY</kbd>
⦿  Command Prefix `]]`

4️⃣ **Star search**
⦿  Hotkey <kbd>EMPTY</kbd>
⦿  Command Prefix `]]]`

### Hotkey Navigation

⚙️ Starting Point → <kbd>CMD+]</kbd>
⚙️ Next Modal 1 → Hit `]`
⚙️ Next Modal 2→ Hit `]]`
⚙️ Next Modal 3→ Hit `]]]`

💡This navigation in important because after I hit <kbd>CMD+]</kbd> to open the first AQS Modal, my right hand is already placed in the key <kbd>]</kbd> .

💡This way, I just have to hit <kbd>]</kbd> again, again and again to be able to open different AQS Modals

💡 It's fast  and let me create groups of AQS modals to blue able to fast switch between notes. 

## THE PROBLEM

When I tried to implement this workflow, I face an issue with the `Command Prefix` text field.

`Command Prefix` text field does not recognize AQS Modals that has repeated letters. this way, I couldn't complete my navigation routine:

✅1️⃣ **Recent Search**
⦿  Hotkey <kbd>CMD+]</kbd>
⦿  Command Prefix `EMPTY`

✅2️⃣ **File name search**
⦿  Hotkey <kbd>EMPTY</kbd>
⦿  Command Prefix `]`

❌3️⃣ **Landmark search**
⦿  Hotkey <kbd>EMPTY</kbd>
⦿  Command Prefix `]]`

❌4️⃣ **Star search**
⦿  Hotkey <kbd>EMPTY</kbd>
⦿  Command Prefix `]]]`

After hitting Command Prefix `]` I was not able to get the next AQS Modal when hitting Command Prefix `]]` 

## MY FR

All said, my FR is about changing the behavior of the `Command Prefix` to be able to recognize repeat letters 🙏🙏🙏

---

Thank for reading this !!!!

I hope it is an feasible change 🙏

I really hope you like my FR this is an important piece if my workflow I have a set of 5 groups of AQS modals that will be triggered from different starting points hotkeys 🙏🙏🙏

Thanks for reading this @tadashi-aikawa I wish you a fantastic day ☀️

**Comment by tadashi-aikawa** [comment]
Hi, @FelipeRearden :)

It would work as you expected if you sort commands as followings.

- 4️⃣ Star search
- 3️⃣ Landmark search
- 2️⃣ File name search
- 1️⃣ Recent Search

Could you please give it a try?

**Comment by FelipeRearden** [comment]
> Hi, @FelipeRearden :)
> 
> It would work as you expected if you sort commands as followings.
> 
> * 4️⃣ Star search
> * 3️⃣ Landmark search
> * 2️⃣ File name search
> * 1️⃣ Recent Search
> 
> Could you please give it a try?

Hello @tadashi-aikawa !!!!

Let me explain more about this .... to make sure we are on the same page...

I use the standard AQS Modals (Star search, Landmark search, File name search, Recent Search) as an EXAMPLE to explain my FR. It is easier to explain this way :)

My real workflow for this are **CUSTOM AQS** Modals created by me to navigate inside my Folder structure :)

If you create new AQS Modals and apply the command prefix criteria that I described above, you will see that we cant use  repeated letters as Command Prefix.

Does it happens on your end as well ?

Thanks for the reply, this is an important FR for me  🙏🙏🙏

**Comment by FelipeRearden** [comment]
⚠️ **UPDATE**  ⚠️

@tadashi-aikawa I applied the rule described from you in TWO Groups of my Custom AQS Modals and it worked perfect !!!!!

The Rule of Thumb here is to let the first letter in the end .... for example:

**AQS Modals ORDER...**

4️⃣ Custom Modal 4
⦿ Hotkey EMPTY
⦿ Command Prefix ]]]

3️⃣ Custom Modal 3
⦿ Hotkey EMPTY
⦿ Command Prefix ]]

2️⃣ Custom Modal 2
⦿ Hotkey EMPTY
⦿ Command Prefix ]

1️⃣ Custom Modal 1 (starting point)
⦿ Hotkey CMD+]
⦿ Command Prefix EMPTY

---

In other words ... applying these rule solved the issue related to this FR. Feel absolutely free to close this issue. 

Thank you very much for the help, I am very very very happy to be able to implement this workflow. 

I wish you a fantastic day ☀️

**Comment by tadashi-aikawa** [comment]
@FelipeRearden 
I’m glad to hear that!

This issue is not about 'the same characters repeating', but 'which condition gets the highest priority when there are multiple matching prefixes'. The one that appears first gets the highest priority."

**Comment by FelipeRearden** [comment]
> @FelipeRearden 
> 
> I’m glad to hear that!

💙

> This issue is not about 'the same characters repeating', but 'which condition gets the highest priority when there are multiple matching prefixes'. The one that appears first gets the highest priority."

Thanks for the info @tadashi-aikawa . 100% understood I am gonna follow this rule when creating my AQS Modals

Have a great day ☀️
